OPUNetGameProtocol.h

```diff
@@ -11,28 +11,34 @@ extern HINSTANCE hInstance;
 
 class OPUNetGameProtocol : public NetGameProtocol
 {
-public:
-	// Member variables
-	OPUNetGameSelectWnd opuNetGameSelectWnd;
-
 public:
 	// Virtual member functions
-	virtual bool IsEnabled() override {return 1;};
-	virtual bool DoStart() override
-	{
-		int retVal;
+
+	virtual bool IsEnabled() override {
+		return 1;
+	};
+
+	virtual bool DoStart() override {
+		// Construct game select window
+		OPUNetGameSelectWnd opuNetGameSelectWnd;
 
 		// Enable Skinning
 		app.mainWnd->PreCreateDlg();
 		// Create the multiplayer setup dialog
-		retVal = opuNetGameSelectWnd.DoModal(MAKEINTRESOURCE(IDD_MultiSetupDialog), hInstance);
+		int retVal = opuNetGameSelectWnd.DoModal(MAKEINTRESOURCE(IDD_MultiSetupDialog), hInstance);

```

Comment:
When executing this line of code, the following exception is thrown:

Exception thrown at 0x14431A80 in Outpost2.exe: 0xC0000005: Access violation executing location 0x14431A80.

Reply:
Ok, based on the address, I think this relates to:
```asm
ExportVirt 0x00431A80, ?DoModal@IDlgWnd@OP2Internal@@UAEHPBDPAUHINSTANCE__@@@Z
```

It looks like the address of `DoModal` with the module base address added to it.

I think what's happening, is because it knows the exact object type in this method, it's direct calling the method rather than going through the virtual function table.